tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -152,25 +152,30 @@ public void CloseSpan(Span span)
             ArraySegment<Span> spansToWrite = default;
 
             // Propagate the resource name to the profiler for root web spans
-            if (span is { IsRootSpan: true, Type: SpanTypes.Web })
+            if (span.IsRootSpan)
             {
-                Profiler.Instance.ContextTracker.SetEndpoint(span.RootSpanId, span.ResourceName);
-
-                var iastInstance = Iast.Iast.Instance;
-                if (iastInstance.Settings.Enabled)
+                if (span.Type == SpanTypes.Web)
                 {
-                    if (_iastRequestContext is { } iastRequestContext)
-                    {
-                        iastRequestContext.AddIastVulnerabilitiesToSpan(span);
-                        iastInstance.OverheadController.ReleaseRequest();
-                    }
-                    else
+                    Profiler.Instance.ContextTracker.SetEndpoint(span.RootSpanId, span.ResourceName);
+
+                    var iastInstance = Iast.Iast.Instance;
+                    if (iastInstance.Settings.Enabled)
                     {
-                        IastRequestContext.AddIastDisabledFlagToSpan(span);
+                        if (_iastRequestContext is { } iastRequestContext)
+                        {
+                            iastRequestContext.AddIastVulnerabilitiesToSpan(span);
+                            iastInstance.OverheadController.ReleaseRequest();
+                        }
+                        else
+                        {
+                            IastRequestContext.AddIastDisabledFlagToSpan(span);
+                        }
                     }
+
+                    _appSecRequestContext?.CloseWebSpan(Tags, span);
                 }
 
-                _appSecRequestContext?.CloseWebSpan(Tags, span);
+                _appSecRequestContext?.DisposeAdditiveContext();
```

Comment:
Should we call DisposeAdditiveContext(); in non web scenarios?

Reply:
yes, I checked, and normally, it should only have been opened in web scenarios, so changed in [475fb54](https://github.com/DataDog/dd-trace-dotnet/pull/6609/commits/475fb54cc4ac23d67f49bd2ac953718aa53546f9)